internal/local/structure_v2.go

```diff
@@ -331,3 +327,204 @@ func parseSync(rootDir string) (*SyncStructure, error) {
+	}
+	if auth.Providers != nil {
+		dataProvider, err := MarshalJSON(auth.Providers)
+		if err != nil {
+			return err
+		}
+		err = WriteFile(filepath.Join(rootDir, NameAuth, FileProviders.String()), 0666, bytes.NewReader(dataProvider))
+		if err != nil {
+			return err
+		}
+	}
+	if auth.CustomUserData != nil {
+		dataCustomUser, err := MarshalJSON(auth.CustomUserData)
+		if err != nil {
+			return err
+		}
+		err = WriteFile(filepath.Join(rootDir, NameAuth, FileCustomUserData.String()), 0666, bytes.NewReader(dataCustomUser))
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func writeSync(rootDir string, sync *SyncStructure) error {
+	if sync == nil || sync.Config == nil {
+		return nil
+	}
+	data, err := MarshalJSON(sync.Config)
+	if err != nil {
+		return err
+	}
+	err = WriteFile(filepath.Join(rootDir, NameSync, FileConfig.String()), 0666, bytes.NewReader(data))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func writeDataSources(rootDir string, dataSources []DataSourceStructure) error {
+	if err := os.MkdirAll(filepath.Join(rootDir, NameDataSources), os.ModePerm); err != nil {
+		return err
+	}
+	for _, ds := range dataSources {
+		nameDataSource := ds.Config["name"].(string)
+		dataDataSource, err := MarshalJSON(ds.Config)
+		if err != nil {
+			return err
+		}
+		err = WriteFile(filepath.Join(rootDir, NameDataSources, nameDataSource, FileConfig.String()), 0666, bytes.NewReader(dataDataSource))
+		if err != nil {
+			return err
+		}
+		for _, rule := range ds.Rules {
+			schema := rule[NameSchema]
+			dataSchema, err := MarshalJSON(schema)
+			if err != nil {
+				return err
+			}
+			delete(rule, NameSchema)
+			dataRule, err := MarshalJSON(rule)
+			if err != nil {
+				return err
+			}
+			err = WriteFile(filepath.Join(rootDir, NameDataSources, nameDataSource, fmt.Sprintf("%s", rule["database"]), fmt.Sprintf("%s", rule["collection"]), FileRules.String()), 0666, bytes.NewReader(dataRule))
+			if err != nil {
+				return err
+			}
+			err = WriteFile(filepath.Join(rootDir, NameDataSources, nameDataSource, fmt.Sprintf("%s", rule["database"]), fmt.Sprintf("%s", rule["collection"]), FileSchema.String()), 0666, bytes.NewReader(dataSchema))
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func writeHTTPEndpoints(rootDir string, httpEndpoints []HTTPEndpointStructure) error {
+	if err := os.MkdirAll(filepath.Join(rootDir, NameHTTPEndpoints), os.ModePerm); err != nil {
+		return err
+	}
+	for _, httpEndpoint := range httpEndpoints {
+		nameHTTPEndpoint := httpEndpoint.Config["name"].(string)
+		data, err := MarshalJSON(httpEndpoint.Config)
+		if err != nil {
+			return err
+		}
+		err = WriteFile(filepath.Join(rootDir, NameHTTPEndpoints, nameHTTPEndpoint, FileConfig.String()), 0666, bytes.NewReader(data))
+		if err != nil {
+			return err
+		}
+		for _, wh := range httpEndpoint.IncomingWebhooks {
+			dataSrc := wh[NameSource].(string)
+			nameWebhook := wh["name"].(string)
+			delete(wh, NameSource)
```

Comment:
likewise to make sure we're not mutating the `AppData` here

Reply:
Yes, making temp versions in next commit.